mantidproject/mantid
# Pulled STL file code out of LoadSampleShape

**Description of work.**
Pulled all code relating to reading STL files out of LoadSampleShape and placed it in its own class so that it can be easily used by other algorithms (in preparation for loading sample environments from stl files.)

**To test:**
Run unit tests for LoadSampleShape, LoadBinaryStl and LoadAsciiStl and ensure they all pass.
Fixes #23621


*This does not require release notes* because **All the changes are internal and do not effect the user.**


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
